design/design_heatmap_norm_counts.Rmd

```diff
@@ -0,0 +1,97 @@
+---
+title: "Design: Add Heatmap Depicting (normalized) Counts"
+author: "Alex Richardson"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+## Objective
+
+A plot method for `AnyHermesData` objects to produce following plots in default setting: 
+1. Generate a heatmap of a single assay, data normalized
+2. Title based on input dataset
+3. How should the function behave if there are several assays?
+4. Organize by assay name or other parameters?
+5. Use filter/subset function to remove genes that are zero value for all samples? 
+  a. Use subset method of hermes package to filter by low expression using subset/rowData
+  b. Ultimate function will have arguments for "filter tech failure, low expr" etc
+6. Review differential expression for relevance to heatmap taxonomy
+7. Collapse by column groups in dataset, eg country, indication, etc?
+8. Colored bands (annotations) above dendrogram, with key or labels - or label dendrogram?
+
+
+## Idea
+
+```{r Heatmap normalize concept}
+library(hermes)
+library(dplyr)
+library(ComplexHeatmap)
+
+# result <- (assays(normalize(hermes_data))[[1]])[1:20,] %>%
+  # subset(a, subset = low_expression_flag, select = DISCSTUD == "N")
+
+result <- hermes_data %>%
+  normalize(methods = 'voom') %>%
+  add_quality_flags() %>%
+  # subset(subset = !low_expression_flag) %>%
+  assays()
+
+result <- result[[1]]
+dim(result)
+
+result
+# We subset from "not low expression" to remove the genes that don't show up in the samples, but we don't see the rows of full zeroes disappearing
+
+notZeroVector <- rowSums(result)
+notZeroVector <- notZeroVector > mean(notZeroVector)
+result <- result[notZeroVector,]
+
+dim(result)
+length(notZeroVector)
+
+Heatmap_plot <- ComplexHeatmap::Heatmap(result)
+draw(Heatmap_plot)
+
+```
+
+```{r Prototype}
+
+NormalizedHeatmap <- function(datasetInput,
```

Comment:
for the name of the function, have a look at the other plotting functions in the `hermes` package to see a pattern